### PR TITLE
Add tests for bluetooth interface

### PR DESCRIPTION
**Pull request by LoonyBean**
Test bluetooth against spec https://webbluetoothcg.github.io/web-bluetooth/#bluetooth

### CODE REVIEW

**Comment by jugglinmike** [comment]
To start off on the right foot, maybe we should include an `OWNERS` file, too.

**Comment by LoonyBean** [comment]
Me myself shouldn't be the owner of Bluetooth (as I work on interopability not Bluetooth), do you know who I should put as owners? 
Thanks

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/214676155)

# Firefox (nightly channel) #
Testing web-platform-tests at revision 3632c1f620f62acc45e0d26e4065ff6d56714d0e
Using browser at version BuildID 20170323110205; SourceStamp 7513b3f42058e9bcf9950d4acf4647d4ad2240f0
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5219/bluetooth/interface.html">/bluetooth/interface.html</a></summary>


|               Subtest                | Results |                  Messages                  |
|--------------------------------------|---------|--------------------------------------------|
|                                      | OK      |                                            |
| `Bluetooth interface prototype test` | FAIL    | `invalid 'in' operand navigator.bluetooth` |

</details>
</details>

**Comment by w3c-bots** [comment]
[View the complete job log.](https://travis-ci.org/w3c/web-platform-tests/jobs/214676156)

# Chrome (unstable channel) #
Testing web-platform-tests at revision 3632c1f620f62acc45e0d26e4065ff6d56714d0e
Using browser at version 59.0.3047.0 dev
Starting 10 test iterations
All results were stable
## All results ##
<details>
<summary>1 test ran</summary>
<details>
<summary><a href="http://w3c-test.org/submissions/5219/bluetooth/interface.html">/bluetooth/interface.html</a></summary>


|               Subtest                | Results |                                Messages                               |
|--------------------------------------|---------|-----------------------------------------------------------------------|
|                                      | OK      |                                                                       |
| `Bluetooth interface prototype test` | FAIL    | `Cannot use 'in' operator to search for 'requestDevice' in undefined` |

</details>
</details>

**Comment by jugglinmike** [comment]
I do not. You might start by asking the specification editors if they'd like to help out.

To be clear: I'm not sure if this is strictly necessary. That said, it seems wise to designate an authority on this topic who can provide reviews before we commit to maintaining tests.

On a more technical note, it appears that the new file has trailing whitespace. This is causing the tests to fail.

**Comment by w3c-bots** [comment]
These tests are now available on [w3c-test.org](https://w3c-test.org/submissions/5219/)

**Comment by LoonyBean** [comment]
Thanks jugglinmike. I agree with you that we should add an OWNERS file, I will look into who to add.

File: bluetooth/interface.html [hunk 1]
```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE html>
+<title>The Bluetooth interface</title>
+<script src="/resources/testharness.js"></script>
+<script src="/resources/testharnessreport.js"></script>
+<script>
+test(function() {
+  assert_true('requestDevice' in navigator.bluetooth);
+  assert_equals(navigator.bluetooth.prototype.requestDevice.length, 0);
```
Comment (foolip) [inline bluetooth/interface.html]: Either `navigator.bluetooth.requestDevice.length` or `Bluetooth.prototype.requestDevice.length`.

**Comment by foolip** [comment]
The module owners in Blink are @jyasskin, @g-ortuno and @scheib. Are y'all interested in owning the wpt directory too?

**Comment by LoonyBean** [comment]
Made changes based on review comments. PTAL

**Review by jyasskin — APPROVED** [review]
@g-ortuno, I assume this is compatible with our plans to upstream the layout tests from https://cs.chromium.org/chromium/src/third_party/WebKit/LayoutTests/bluetooth/?

File: bluetooth/OWNERS [hunk 2]
```diff
@@ -0,0 +1,3 @@
+ @jyasskin,
```
Comment (foolip) [inline bluetooth/OWNERS]: There should be no commas in these files

**Comment by g-ortuno** [comment]
Would it make sense to modify [idl-Bluetooth.html](https://cs.chromium.org/chromium/src/third_party/WebKit/LayoutTests/bluetooth/idl/idl-Bluetooth.html) or [idl-NavigatorBluetooth.html](https://cs.chromium.org/chromium/src/third_party/WebKit/LayoutTests/bluetooth/idl/idl-NavigatorBluetooth.html) to include the length check and then upstream that test?

**Comment by LoonyBean** [comment]
I was thinking about adding it to idl-Bluetooth.html.
I am not sure how upstreaming works, but yeah I think that doing it that
way makes most sense.

**Comment by LoonyBean** [comment]
Upstreaming the idl-Bluetooth.html at: https://codereview.chromium.org/2787103002/

**Pull request closed without merging**